server/controllers/administrativeControllers/universityMajors.js

```diff
@@ -0,0 +1,144 @@
+const User = require('../../models/user');
+const JWT = require('jsonwebtoken');
+const generateRes = require('../../helpers/generateJSON');
+const Major = require('../../models/majors');
+const { adminMajors } = require('../../helpers/inputValidation/validation');
+
+
+/*  This function handles the addition of new major(s) to the database, requires a JWT, should be used with a POST request, and should only
+    be accessible by adminstrative accounts. This will take an array of one or more majors and add them to the database if they
+    do not already exist. 
+
+    This function takes one field in the request body: majors (Array of Strings, the list of majors or major to be added to the db)
+*/
+const addMajors = async (req, res) => {
+    try {
+        //validates the request body
+        const { error } = adminMajors.validate(req.body);
+        if (error) {
+            return res.status(400).json(generateRes(false, 400, "INPUT_ERROR", {
+                errors: error.details,
+                original: error._original
+            }));
+        }
+
+        //decode the user's access token and then decodes it
+        const accessToken = req.header('Authorization').split(' ')[1];
+        const decodeAccessToken = JWT.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+        //grab the adminstrator's account
+        let admin = await User.findOne({ email: decodeAccessToken.email });
+        //check that the account type is an adminstrator account, other throw a authorized error response
+        if (admin.userType.Type == process.env.ADMIN) {
+            let majorsRecord = await Major.findOne({ location: admin.universityLocation });
+            //checks if a major record could be found, if so then adds the majors to the db, otherwise creates a new record.
+            if (!majorsRecord) {
+                let majors = new Major({
+                    location: admin.universityLocation,
+                    majors: req.body.majors
+                })
+                await majors.save();
+            } else {
+                //Get a new array of all elements from the request body that do not exist in the db, then concat them into the db
+                let newMajors = req.body.majors.filter(major => !majorsRecord.majors.includes(major));
```

Comment:
I assume majors is an array, so maybe take a look at the `String.prototype.find()` method instead of filtering for a specific major and seeing if the length of the result is 0